src/main/java/slimeknights/tconstruct/tools/modifiers/shared/NecroticModifier.java

```diff
@@ -13,9 +13,9 @@ public NecroticModifier() {
 
   @Override
   public int afterLivingHit(IModifierToolStack tool, int level, LivingEntity attacker, LivingEntity target, float damageDealt, boolean isCritical, float cooldown) {
-    if (cooldown > 0.9 && damageDealt > 0) {
+    if (cooldown > 0.9 && damageDealt > 0 && target.getHealth() <= 0) {
       // heals between 0 and (level) * 5% of damage dealt
-      float heal = attacker.getRNG().nextFloat() * damageDealt * level * 0.05f;
+      float heal = attacker.getRNG().nextFloat() * damageDealt * level * 0.08f + damageDealt * level * 0.02f;
```

Comment:
Add some parenthesis here, as the order of operations gets a little unclear with such long chains. Math makes sense but its a little unreadable.